app/models/reference_history.rb

```diff
@@ -0,0 +1,68 @@
+class ReferenceHistory
+  Event = Struct.new(:name, :time, :extra_info)
+
+  attr_reader :reference
+
+  def initialize(reference)
+    @reference = reference
+  end
+
+  def all_events
+    request_sent
+      .concat(request_cancelled)
+      .concat(reminder_sent)
+      .concat(request_bounced)
+      .concat(request_declined)
+      .concat(reference_given)
+      .sort_by(&:time)
+  end
+
+  def request_sent
+    audits
+      .select { |a| status_change(a, to: 'feedback_requested') }
+      .map { |a| Event.new('request_sent', a.created_at) }
+  end
+
+  def request_cancelled
+    audits
+      .select { |a| status_change(a, to: 'cancelled') }
```

Comment:
Should we include the `cancelled_at_end_of_cycle` status here?

Reply:
Good question. @paulrobertlloyd - should we have a 'cancelled at end of cycle' event in the reference history?

Reply:
I think so. The request status should also say cancelled (and the reason why, although that’s currently missing, pending design/content), but I don’t think it hurts to repeat this in the history.